_blog-src/_posts/2016-04-08-data-encryption.adoc

```diff
@@ -0,0 +1,142 @@
+    thing = config.get("thing"); //$NON-NLS-1$
+
+    if (thing == null) {
+      throw new RuntimeException("Missing configuration property: apiman-manager.config.thing"); //$NON-NLS-1$
+    }
+  }
+}
+```
+
+
+Here, we are passing a configuration object that gets created in the `ConfigFactory` we discussed above. Once the `SuperDataEncrypter` class is invoked, the object is checked for a "thing" key. If it is null, a runtime exception is thrown. Pretty, straightforward so far, right?
+
+When saving data into Elasticsearch, the data is _pre-marshalled_ and _post-marshalled_, which checks to see if you have a _pre-encryption_ and _post-encryption_ method in your class, respectively. You can pretty much guess what our next step will be:
+
+```java
+public class SuperDataEncrypter implements IDataEncrypter {
+  /**
+  * Constructor.
+  */
+  public SuperDataEncrypter(Map<String, String> config) {
+    thing = config.get("thing"); //$NON-NLS-1$
+
+    if (thing == null) {
+      throw new RuntimeException("Missing configuration property: apiman-manager.config.thing"); //$NON-NLS-1$
+    }
+  }
+
+
+  /* (non-Javadoc)
+   * @see io.apiman.common.util.crypt.IDataEncrypter#encrypt(java.lang.String)
+   */
+  @Override
+  public String encrypt(String plainText) {
+      return AesEncrypter.encrypt(secretKey, plainText);
+  }
+
+  /* (non-Javadoc)
+   * @see io.apiman.common.util.crypt.IDataEncrypter#decrypt(java.lang.String)
+   */
+  @Override
+  public String decrypt(String encryptedText) {
+      return AesEncrypter.decrypt(secretKey, encryptedText);
+  }
+}
+```
+
+From there, you can use your `SuperDataEncrypter` class to write the functionality you need. For an example of an implementation you can take a look at the `AesDataEncrypter` class we discussed earlier. You should be up and running in no time!
+
+NOTE: If you are kind enough to want to share your plugin with the rest of the apiman community, create a pull request on the main https://github.com/apiman/apiman[apiman repository]. Contributors, please don't forget to write a unit test for any plugin you write (one that actually passes, kthx!).
+
+
+== Key Takeaways
+
+If your memory is as short-lived as mine, it’s probably best that we do a quick recap. In this article we discussed the importance of securing stored data, and why it should be treated as an essential component of your API security strategy. We then discussed how apiman encrypts and decrypts sensitive data, with a detailed review of the IDataEncrypter interface and the classes that implement it.
+
+For those of us with a guilty security conscience, we went over how to configure the API gateway to use an encrypted key that you provided in the `apiman.properties` file. Because we’re über ambitious, we also took a look at how to write your very own implementation of said `IDataEncrypter` interface and the methods you are required to include when using Elasticsearch as a data store.
```

Comment:
Suggest you replace "include when using Elasticsearch as a data store." with: 

implement.